5calls/app/src/androidTest/java/org/a5calls/android/a5calls/controller/MainActivityHappyPathTest.java

```diff
@@ -0,0 +1,406 @@
+        AppSingleton.getInstance(
+                InstrumentationRegistry.getInstrumentation().getTargetContext())
+                .setRequestQueue(requestQueue);
+
+        // Create a new FiveCallsApi with our mock RequestQueue
+        String callerId = AccountManager.Instance.getCallerID(
+                InstrumentationRegistry.getInstrumentation().getTargetContext());
+        FiveCallsApi api = new FiveCallsApi(callerId, requestQueue);
+        AppSingleton.getInstance(
+                InstrumentationRegistry.getInstrumentation().getTargetContext())
+                .setFiveCallsApi(api);
+
+        // Create an idling resource to wait for the UI to load
+        final ConditionIdlingResource idlingResource = new ConditionIdlingResource();
+        IdlingRegistry.getInstance().register(idlingResource);
+
+        try {
+            // Set up the mock to handle all possible requests with appropriate responses
+            mHttpStack.clearUrlPatternResponses();
+            mHttpStack.setResponseForUrlPattern("issues", issuesResponse);
+            mHttpStack.setResponseForUrlPattern("reps", contactsResponse);
+            mHttpStack.setResponseForUrlPattern("report", reportResponse);
+
+            // Set a default response for any other requests
+            mHttpStack.setResponseToReturn(new HttpResponse(200, new ArrayList<>(), "{}".getBytes()));
+
+            // Launch the activity
+            ActivityScenario<MainActivity> scenario = ActivityScenario.launch(MainActivity.class);
+
+            // Wait for all requests to complete and UI to update
+            try {
+                Thread.sleep(3000);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+
+            // Set the idling resource to idle
+            idlingResource.setIdle(true);
+
+            // Verify that the toolbar is displayed
+            onView(withId(R.id.toolbar)).check(matches(isDisplayed()));
+
+            // Verify that the issues recycler view is displayed
+            onView(withId(R.id.issues_recycler_view)).check(matches(isDisplayed()));
+
+            // Verify that the filter spinner is displayed
+            onView(withId(R.id.filter)).check(matches(isDisplayed()));
+
+            // Verify that our test issue is displayed
+            onView(withText("Test Issue 1")).check(matches(isDisplayed()));
+
+            // The subtitle might not contain the exact location string, so let's just verify it's displayed
+            onView(withId(R.id.action_bar_subtitle)).check(matches(isDisplayed()));
+
+            // Close the activity
+            scenario.close();
+        } finally {
+            // Unregister the idling resource
+            IdlingRegistry.getInstance().unregister(idlingResource);
+        }
+    }
+
+    @Test
+    public void testNavigationDrawerOpens() {
+        // Set up mock to return valid responses instead of empty ones
+        try {
```

Comment:
so much repeated code!! try to share some -- should get cleaner after you move JSON object stuff to fakeJSONData.